web/models/class/class_helper.ex

```diff
@@ -38,6 +39,42 @@ defmodule CoursePlanner.ClassHelper do
   end
 
   defp non_deleted_query do
-    from c in Class , where: is_nil(c.deleted_at)
+    from c in Class,
+      preload: [{:offered_course, :course}],
+      where: is_nil(c.deleted_at),
+      order_by: [desc: :date, desc: :starting_at]
+  end
+
+  defp get_class_students(offered_course_id) do
+    Repo.all(
+    from c in Class,
+      join: oc in assoc(c, :offered_course),
+      preload: [:students, offered_course: oc],
+      where: oc.id == ^offered_course_id
+      )
+  end
+
+  def create_class_attendance_records(class) do
+    class_data = List.first(get_class_students(class.offered_course_id))
```

Comment:
now you can access the students of class with `class.students` if you preload the students association

are you sure you even need to call this method?